### ISSUE TITLE
403 on path-from-root while there is a path

### ISSUE THREAD

**Issue by smadbe** [France-ioi/AlgoreaBackend#850]
## Motivations
Fix problem described below (see "Issue")

## Subtasks
Update the itemPathFromRootFind service so that
* the rule about chain of attempts becomes: " The chain of attempts of the path cannot have missing results for items requiring explicit entry FOR ANY PATH ELEMENT EXCEPT FOR THE LAST ONE or not started results
within or below ended/not-allowing-submissions attempts. "
* fix the implementation to reflect this change

# Issue

The "Find an item path" service state "The path consists only of the items visible to the participant (can_view>='content' for all the items except for the last one and can_view>='info' for the last one)".

However, as a temp user:
```
curl 'https://dev.algorea.org/api/items/4297212147988794347/path-from-root' -H 'Cookie: access_token=3!pru7gasyd7a2zb9hpalc185pl84w73qh!dev.algorea.org!/api/' 
{"success":false,"message":"Forbidden","error_text":"Insufficient access rights"}
```
While:
```
$ curl 'https://dev.algorea.org/api/items/4702/1625159049301502151/4297212147988794347/breadcrumbs?parent_attempt_id=0'  -H 'Cookie: access_token=3!pru7gasyd7a2zb9hpalc185pl84w73qh!dev.algorea.org!/api/' 
[{"attempt_id":"0","item_id":"4702","language_tag":"fr","title":"Parcours officiels","type":"Chapter"},{"attempt_id":"0","item_id":"1625159049301502151","language_tag":"fr","title":"MOTIF ART","type":"Chapter"},{"item_id":"4297212147988794347","language_tag":"en","title":"Sur le fil","type":"Task"}]
```
and perm on `4702`, `1625159049301502151` and `4297212147988794347` are respectively `content`, `content` and`info`

**Comment by GeoffreyHuck** [comment]
Two questions about how this change should  be reflected in `resultStartPath`?

From my understanding, `resultStartPath` gets for each item in the path:
- An `attempt_id`
- Whether it has a started result: `has_started_result`. If it doesn't, a result is added in the DB and a results propagation is done.

* With the change, the last `item` can have `requires_explicit_entry=1`, so it's possible there is no `attempt`. What should we do in this case?

* Also, the response of `resultStartPath` contains an `attempt_id`. It is the `attempt_id` associated with the last item in the path. What do we return in this case?

**Comment by smadbe** [comment]
An important aspect I've just realized rechecking all doc and the frontend implem:
- itemPathFromRootFind returns the queried element. So `/items/70/path-from-root` returns `["10", "60", "70"]`. So I use it when I want to find the path of "70" and so I don't care about the constraints (visibility >= "content",  requires_explicit_entry=1) on 70.
- resultStartPath starts all the path. So typically in the same case, when I have found `["10", "60", "70"]` , I call `/items/10/60/start-result-path` (so not including `70`). Then, when the "path is started", I fall down to the regular case where I check if/how I start item 70 but that's not related with these services anymore.

So the "don't require `visibility >= "content",  requires_explicit_entry=1` on the last element is only valid for `itemPathFromRootFind`, not for `resultStartPath`.

> With the change, the last item can have requires_explicit_entry=1, so it's possible there is no attempt. What should we do in this case?

For the start path, it should still be an error.

> Also, the response of resultStartPath contains an attempt_id. It is the attempt_id associated with the last item in the path. What do we return in this case?

No change of behavior.
